File: annotations/ts/src/images.ts
export type ImageMeta = {
    box_contours?:      Array<Array<Array<number[]>>> | null;
    grid_coords?:       null | Rectangle;
    robust_extraction?: RobustExtraction | null;
    source?:            Source | null;
    [property: string]: any;
}


export type RobustExtraction = "failed" | "incorrect" | "perspective-correct" | "correct";

export type Source = "raw-scan" | "corrected-scan" | "camera" | "corrected-camera" | "robust-corrected";

  // hand-generated as quicktype treats `[number, number]` as `any[]`
  export type Rectangle = {
    size: [number, number]
    tl: [number, number]
  }
  

File: annotations/src/moveread/annotations/images/images.py
from typing import Literal
from pydantic import ConfigDict
from moveread.boxes import Annotations
from robust_extraction.manual._correct import Corners

Source = Literal['raw-scan', 'corrected-scan', 'camera', 'corrected-camera', 'robust-corrected'] 

class ImageMeta(Annotations):
  model_config = ConfigDict(extra='allow')
  source: Source | None = None
  perspective_corners: Corners | None = None
